### PR TITLE
refactor: use unplugin-icons to unify all icons

### PR DESCRIPTION
**Pull request by mancuoj**
将图标统一使用 unplugin-icons 引入，移除了两个相关的图标库，也方便之后的图标引入

### CODE REVIEW

**Comment by Kaiyiwing** [comment]
正在进行中的 pr 可以设置为 draft pr

File: src/pages/Gallery-N/index.tsx [hunk 1]
```diff
@@ -8,12 +8,12 @@ import { currentDictInfoAtom } from '@/store'
 import { Dictionary, LanguageCategoryType } from '@/typings'
 import groupBy, { groupByDictTags } from '@/utils/groupBy'
 import * as ScrollArea from '@radix-ui/react-scroll-area'
-import { IconX } from '@tabler/icons-react'
 import { useAtomValue } from 'jotai'
 import { createContext, useCallback, useEffect, useMemo } from 'react'
 import { useHotkeys } from 'react-hotkeys-hook'
 import { useNavigate } from 'react-router-dom'
 import { Updater, useImmer } from 'use-immer'
+import TablerX from '~icons/tabler/x'
```
Comment (chengluyu) [inline src/pages/Gallery-N/index.tsx:16]:
```suggestion
import IconX from '~icons/tabler/x'
```

File: src/pages/Gallery-N/index.tsx [hunk 2]
```diff
@@ -66,7 +66,7 @@ export default function GalleryPage() {
       <GalleryContext.Provider value={{ state: galleryState, setState: setGalleryState }}>
         <ChapterList />
         <div className="relative mb-auto mt-auto flex w-full flex-1 flex-col overflow-y-auto pl-20">
-          <IconX className="absolute right-20 top-10 mr-2 h-7 w-7 cursor-pointer text-gray-400" onClick={onBack} />
+          <TablerX className="absolute right-20 top-10 mr-2 h-7 w-7 cursor-pointer text-gray-400" onClick={onBack} />
```
Comment (chengluyu) [inline src/pages/Gallery-N/index.tsx:69]:
```suggestion
          <IconX className="absolute right-20 top-10 mr-2 h-7 w-7 cursor-pointer text-gray-400" onClick={onBack} />
```

File: src/pages/Gallery-N/index.tsx (hunk 1 above)
Comment (chengluyu) [reply src/pages/Gallery-N/index.tsx:16]: TablerX is a confusing name.

File: src/pages/Gallery/ChapterButton.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { useChapterStats } from './hooks/useChapterStats'
 import useIntersectionObserver from '@/hooks/useIntersectionObserver'
-import { CheckCircleIcon } from '@heroicons/react/20/solid'
 import React, { useEffect, useRef } from 'react'
+import HeroiconsCheckCircle from '~icons/heroicons/check-circle-solid'
```
Comment (chengluyu) [inline src/pages/Gallery/ChapterButton.tsx:4]:
```suggestion
import IconCheckCircle from '~icons/heroicons/check-circle-solid'
```

File: src/pages/Gallery/ChapterButton.tsx [hunk 4]
```diff
@@ -50,7 +50,7 @@ export const ChapterButton: React.FC<ChapterButtonProps> = ({ index, selected, w
       )}
 
       {selected ? (
-        <CheckCircleIcon className="absolute -bottom-4 -right-4 h-18 w-18 text-6xl text-green-500 opacity-60 dark:text-green-300" />
+        <HeroiconsCheckCircle className="absolute -bottom-4 -right-4 h-18 w-18 text-6xl text-green-500 opacity-60 dark:text-green-300" />
```
Comment (chengluyu) [inline src/pages/Gallery/ChapterButton.tsx:53]:
```suggestion
        <IconCheckCircle className="absolute -bottom-4 -right-4 h-18 w-18 text-6xl text-green-500 opacity-60 dark:text-green-300" />
```

File: src/pages/Gallery/ChapterButton.tsx (hunk 3 above)
Comment (chengluyu) [reply src/pages/Gallery/ChapterButton.tsx:4]: Same, please don´t include library´s name in the component.

File: vite.config.ts [hunk 5]
```diff
@@ -12,7 +13,20 @@ export default defineConfig(async () => {
...
...,21.15,0,31.72,0,0,4.64,0,9.29,0,13.93-2.91,0-5.81,0-8.72,0q0,20.89,0,41.77c4.45.05,8.91,0,13.36.05q0,6.94,0,13.9c-15.48,0-31,0-46.44,0,2.09-4.64,4.19-9.28,6.34-13.9,4.15,0,8.31,0,12.46-.05,0-13.91,0-27.83,0-41.74-2.9,0-5.79,0-8.69,0Q385.55,358.77,385.58,351.81Z" transform="translate(-256 -256)" style="fill:#fff"/><path d="M476.93,353.09c3.5-2.9,9.31-1,10.62,3.27,1.67,3.92-1.46,8.77-5.62,9.15-2.59.23-5.19.1-7.78.09C474.32,361.41,473.05,356,476.93,353.09Z" transform="translate(-256 -256)" style="fill:#fff"/><path d="M283,365.72c4.57,0,9.13.07,13.69-.11-.28,7.81-1.15,15.6-1.64,23.4-.58,9.17-2.66,18.61-8.26,26.1-2.5-5.28-4.84-10.63-7.18-16a22.12,22.12,0,0,0,1.32-6.06C281.56,384,282.27,374.84,283,365.72Z" transform="translate(-256 -256)" style="fill:#fff"/><path d="M329.1,365.72q6.91,0,13.83,0c.7,9.09,1.42,18.18,2.08,27.27a21.67,21.67,0,0,0,1.32,6.15c-2.34,5.35-4.7,10.69-7.17,16-5.18-6.91-7.37-15.55-8.1-24S329.77,374.18,329.1,365.72Z" transform="translate(-256 -256)" style="fill:#fff"/><path d="M346.26,420.41q3.17-7,6.4-13.95c7.13,1.83,14.53.79,21.8,1.07-2.08,4.66-4.21,9.3-6.35,13.93C360.84,421.07,353.34,422.5,346.26,420.41Z" transform="translate(-256 -256)" style="fill:#fff"/></svg>',
```
Comment (chengluyu) [inline vite.config.ts:25]: Please load the SVG from files. For example,

```
  return {
    plugins: [
      react({ babel: { plugins: [jotaiDebugLabel, jotaiReactRefresh] } }),
      visualizer() as PluginOption,
      Icons({
        compiler: 'jsx',
        jsx: 'react',
        customCollections: {
          'my-icons': {
            xiaohongshu: await fs.readFile("xiaohongshu.svg", { encoding: "utf-8" }),
```

File: src/pages/Typing/components/SoundSwitcher/index.tsx [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import { keySoundsConfigAtom, hintSoundsConfigAtom } from '@/store'
 import { Popover, Transition, Switch } from '@headlessui/react'
-import { SpeakerWaveIcon } from '@heroicons/react/24/solid'
 import { useAtom } from 'jotai'
 import { Fragment, useCallback } from 'react'
+import HeroiconsSpeakerWave from '~icons/heroicons/speaker-wave-solid'
```
Comment (chengluyu) [inline src/pages/Typing/components/SoundSwitcher/index.tsx:5]:
```suggestion
import IconSpeakerWave from '~icons/heroicons/speaker-wave-solid'
```

File: src/pages/Typing/components/SoundSwitcher/index.tsx [hunk 7]
```diff
@@ -36,7 +36,7 @@ export default function SoundSwitcher() {
             aria-label="音效设置"
             title="音效设置"
           >
-            <SpeakerWaveIcon className="icon" />
+            <HeroiconsSpeakerWave className="icon" />
```
Comment (chengluyu) [inline src/pages/Typing/components/SoundSwitcher/index.tsx:39]:
```suggestion
            <IconSpeakerWave className="icon" />
```

File: src/pages/Typing/components/Switcher/index.tsx [hunk 8]
```diff
@@ -3,11 +3,17 @@ import Setting from '../Setting'
...
 import { useContext } from 'react'
 import { useHotkeys } from 'react-hotkeys-hook'
+import HeroiconsEyeSlash from '~icons/heroicons/eye-slash-solid'
+import HeroiconsEye from '~icons/heroicons/eye-solid'
+import HeroiconsMoon from '~icons/heroicons/moon-solid'
+import HeroiconsSun from '~icons/heroicons/sun-solid'
+import TablerLanguage from '~icons/tabler/language'
+import TablerLanguageOff from '~icons/tabler/language-off'
+import TablerRepeatOff from '~icons/tabler/repeat-off'
+import TablerRepeatOnce from '~icons/tabler/repeat-once'
```
Comment (chengluyu) [inline src/pages/Typing/components/Switcher/index.tsx:16]: Same here. Please do not include library's name in component names.

File: src/pages/Typing/components/ShareButton/index.tsx [hunk 9]
```diff
@@ -1,8 +1,8 @@
 import SharePicDialog from './SharePicDialog'
 import { recordShareAction } from '@/utils'
 import { flip, offset, shift, useFloating, useHover, useInteractions, useRole } from '@floating-ui/react'
-import { IconShare2 } from '@tabler/icons-react'
 import { useCallback, useMemo, useState } from 'react'
+import TablerShare2 from '~icons/tabler/share-2'
```
Comment (chengluyu) [inline src/pages/Typing/components/ShareButton/index.tsx:5]: Same here. Please do not include library's name in component names.

File: src/pages/Typing/components/ShareButton/SharePicDialog.tsx [hunk 10]
```diff
@@ -12,9 +12,9 @@ import keyboardSvg from '@/assets/sharePic/keyBackground.svg'
 import { currentChapterAtom, currentDictInfoAtom } from '@/store'
 import { recordShareAction } from '@/utils'
 import { Dialog, Transition } from '@headlessui/react'
-import { XMarkIcon } from '@heroicons/react/24/solid'
 import { useAtomValue } from 'jotai'
 import { Fragment, useCallback, useContext, useEffect, useMemo, useRef, useState } from 'react'
+import HeroiconsXMark from '~icons/heroicons/x-mark-solid'
```
Comment (chengluyu) [inline src/pages/Typing/components/ShareButton/SharePicDialog.tsx:17]: Same here. Please do not include library's name in component names.

File: src/pages/Typing/components/Setting/index.tsx [hunk 11]
```diff
@@ -3,10 +3,13 @@ import AdvancedSetting from './AdvancedSetting'
...
 import { Dialog, Tab, Transition } from '@headlessui/react'
-import { Cog6ToothIcon } from '@heroicons/react/24/solid'
-import { IconEar, IconAdjustmentsHorizontal, IconX, IconDatabaseCog } from '@tabler/icons-react'
 import classNames from 'classnames'
 import { Fragment, useContext, useState } from 'react'
+import HeroiconsCog6Tooth from '~icons/heroicons/cog-6-tooth-solid'
+import TablerAdjustmentsHorizontal from '~icons/tabler/adjustments-horizontal'
+import TablerDatabaseCog from '~icons/tabler/database-cog'
+import TablerEar from '~icons/tabler/ear'
+import TablerX from '~icons/tabler/x'
```
Comment (chengluyu) [inline src/pages/Typing/components/Setting/index.tsx:12]: Same here. Please do not include library's name in component names.

File: src/components/Footer/index.tsx [hunk 12]
```diff
@@ -1,18 +1,20 @@
...
-import { IconBrandWechat, IconCoffee, IconTerminal2 } from '@tabler/icons-react'
 import { useAtom } from 'jotai'
 import React, { useCallback } from 'react'
+import MaterialSymbolsMail from '~icons/material-symbols/mail'
+import MdiCoffee from '~icons/mdi/coffee'
+import XiaoHongShu from '~icons/my-icons/xiaohongshu'
+import SimpleIconsGithub from '~icons/simple-icons/github'
+import SimpleIconsVisualstudiocode from '~icons/simple-icons/visualstudiocode'
+import SimpleIconsWechat from '~icons/simple-icons/wechat'
+import TwemojiFlagChina from '~icons/twemoji/flag-china'
```
Comment (chengluyu) [inline src/components/Footer/index.tsx:17]: Please do not include library's name in component names.

File: src/pages/Gallery-N/DictRequest.tsx [hunk 13]
```diff
@@ -1,6 +1,6 @@
 import InfoPanel from '@/components/InfoPanel'
-import { IconBook2 } from '@tabler/icons-react'
 import { useCallback, useState } from 'react'
+import TablerBook2 from '~icons/tabler/book-2'
```
Comment (chengluyu) [inline src/pages/Gallery-N/DictRequest.tsx:3]: Same here. Please do not include the library's name in component names.

**Review by chengluyu — CHANGES_REQUESTED** [review]
Hi. Thank you for your contribution! I found some problems. Please fix them.

**Comment by Kaiyiwing** [comment]
CleanShot 2023-04-28 at 00 21 19@2x

看起来这里 logo 分辨率有点低的样子，而且两侧被裁切掉了

File: src/components/Footer/index.tsx [hunk 14]
```diff
@@ -84,9 +86,9 @@ const Footer: React.FC = () => {
       <InfoPanel
         openState={infoPanelState.community}
         title="用户反馈社群"
-        icon={IconBrandWechat}
+        icon={IconWechat}
         buttonClassName="bg-cyan-500 hover:bg-cyan-400"
-        iconClassName="text-cyan-600 bg-cyan-300"
```
Comment (Kaiyiwing) [inline src/components/Footer/index.tsx:89]: 🤔️ 这个地方把所有 icon 的 bg 移除掉的目的是？

File: src/components/Footer/index.tsx [hunk 15]
```diff
@@ -211,27 +194,27 @@ const Footer: React.FC = () => {
...
         <Tooltip content="中国大陆镜像">
           <a href="https://kaiyiwing.gitee.io/qwerty-learner" target="_self" title="前往中国大陆镜像">
-            <img src={cnFlag} className="h-5 w-5 cursor-pointer" alt="中国国旗" />
+            <IconFlagChina fontSize={16} />
           </a>
         </Tooltip>
 
-        <button
-          className="cursor-pointer text-gray-500 no-underline hover:no-underline dark:text-gray-400 "
+        {/* <button
```
Comment (Kaiyiwing) [inline src/components/Footer/index.tsx:206]: 这里为啥注释掉了

Comment (mancuoj) [reply src/components/Footer/index.tsx:206]: 看起来跟咖啡图标的功能重合了（其实是忘了取消注释，你看是保留还是

File: src/components/Footer/index.tsx (hunk 14 above)
Comment (mancuoj) [reply src/components/Footer/index.tsx:89]: 这里是因为之前的圆形背景在暗黑模式下颜色看起来特别奇怪，上一个问题看起来要大改一下InfoPanel的样式

File: src/components/Footer/index.tsx (hunk 15 above)
Comment (Kaiyiwing) [reply src/components/Footer/index.tsx:206]: 还是保留吧

**Review by Kaiyiwing — APPROVED** [review]

**Pull request merged**